src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -220,7 +220,8 @@ void AppHost::_HandleCommandlineArgs(const Remoting::WindowRequestedArgs& window
         // seemed to reorder bits of init so much that everything broke. So
         // we'll leave it here.
         const auto numPeasants = _windowManager.GetNumberOfPeasants();
-        if (numPeasants == 1)
+        // Don't attempt to session restore if we're just making a window for tear-out
+        if (!startedForContent && numPeasants == 1)

```

Comment:
BTW I have to mention that this code is kind of... weird? I'm honestly not sure how to describe this feeling. Basically, I'm confused why session restore related code runs inside `_HandleCommandlineArgs` why it's related to `numPeasants`. And do we ever have more than 1 peasant nowadays? Not really asking, I'm just a lil puzzled about this code. I suppose that's how it feels like when reading AtlasEngine code for you all... 😄

Reply:
It's super weird that's for sure. All of session restore sorta is. It probably could live in a helper just after `_HandleCommandlineArgs`. Might just make a PR for that for s&g's

Reply:
we may want to take a todo to really fine-toothed-comb this.